dnf-testing.py

```diff
@@ -0,0 +1,351 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import argparse
+import os
+import re
+import subprocess
+import sys
+
+DESTRUCTIVE_TAGS = set([
+    "@destructive",
+    "@no_installroot",
+])
+PROGPATH = os.path.abspath(os.path.dirname(sys.argv[0]))
+
+
+class MultilineHelpFormatter(argparse.HelpFormatter):
+    '''
+    Help formatter that allows multiline help texts.
+    '''
+    def _split_lines(self, text, width):
+        if '\n' in text:
+            return text.splitlines()
+        return super(MultilineHelpFormatter, self)._split_lines(text, width)
+
+def command_line_parser():
+    '''
+    Initialize and return command line arguments parser.
+    '''
+    parser = argparse.ArgumentParser(
+        description="Functional tests for DNF",
+        formatter_class=MultilineHelpFormatter)
+    parser.add_argument(
+        "--featuresdir", default="features",
+        help="Directory with *.feature files")
+    parser.add_argument(
+        "-c", "--container", metavar="IMAGE",
+        default='dnf-bot/dnf-testing:latest',
+        help="Specified Image ID or name if do not want to run the last built image")
+    parser.add_argument(
+        "-d", "--devel", action="store_true", default=False,
+        help="Share local feature/ with docker")
+    parser.add_argument(
+        "-f", "--file", metavar="FILE", dest="docker_file",
+        default="Dockerfile", help="Path to Dockerfile to use")
+    parser.add_argument(
+        "-p", "--podman", action="store_true", default=False,
+        help="Force using podman instead of docker")
```

Comment:
I wonder if we'd like to make podman the default. Granted we've had some issues lately with podman being broken, but otherwise, for development, maybe most of us would prefer to use podman as the default and only fall back to docker if podman isn't present? For CI this shouldn't make a difference?

Reply:
Right, I changed the logic so that the podman is the primary choice and docker is a fallback.